components/notifier/src/strategies/notification_strategy.py

```diff
@@ -0,0 +1,67 @@
+"""Strategies for notifying users about metrics."""
+
+from datetime import datetime, timedelta
+from typing import List, Optional
+from models.notification import Notification
+from models.metric_notification_data import MetricNotificationData
+
+
+class NotableEvents:
```

Comment:
Are instance of this class notable events? I don't think so. The responsibility of this class is to determine notifications, so maybe call it something like "NotificationFinder"?